src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -139,6 +139,21 @@ public function map(callable $callback)
     }
 
     /**
+     * Map a collection and flatten the result by a single level.
+     *
+     * @param  callable  $callback
+     * @return \Illuminate\Support\Collection|static
+     */
+    public function flatMap(callable $callback)
+    {
+        $result = parent::map($callback)->collapse();
+
+        return ! $result->count() || $result->contains(function ($item) {
```

Comment:
What happens when, in your example, there are no users?

Will it now return a support collection? That's dangerous!